pkgs/development/libraries/tiledb/default.nix

```diff
@@ -0,0 +1,32 @@
+{ pkgs ? import ./pkgs.nix }:
+
+with pkgs;
+
+  stdenv.mkDerivation rec {
+    name = "tiledb";
+    version = "1.5.1";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "TileDB-Inc";
+      repo = "TileDB";
+      rev = "7236bb29f2aa1ac4de489f0791e9265cb257184d";
+      sha256 = "sha256:0ky0dcv1w1jn1cjn3819aq9xyd2wg80aagf2flxmd916flgr9zjl";
+  };
+
+  makeTarget = "tiledb";
+  outputs = ["py" "out"];
+
+  nativeBuildInputs = [ doxygen cmake zlib lz4 bzip2 zstd spdlog tbb openssl];
```

Comment:
Can you explain why you separated between `nativeBuildInputs` and `buildInputs` here?

Reply:
I am not sure about this.

Reply:
Well `nativeBuildInputs` usually means the CLI commands are available during build. While `buildInputs` the CLI commands are not. Also both deps only exist during build time. During runtime these do not exist. For ones that must exist at runtime it's `propagatedBuildInputs` or `propagatedNativeBuildInputs`. So basically this should be mostly programs that must RUN during build time.

I think what you have there should be `buildInputs`. Not `nativeBuildInputs`. But `doxygen` looks like it needs to run during build time, so `nativeBuildInputs` instead.